Quaver.Shared/Audio/AudioEngine.cs

```diff
@@ -173,7 +173,7 @@ public static double GetNearestSnapTimeFromTime(Qua map, Direction direction, in
 
             var nearestTick = Math.Round((pointToSnap - point.StartTime) / snapTimePerBeat) * snapTimePerBeat + point.StartTime;
 
-            if ((int) Math.Abs(nearestTick - time) <= (int) snapTimePerBeat)
+            if ((int)Math.Abs(nearestTick - time) <= (int)snapTimePerBeat || Math.Abs(nearestTick - time) - snapTimePerBeat <= 0.1)
```

Comment:
So I'm not exactly quite sure what this condition is doing (why is it rounding to ints?), but either way this seems like an odd way to correct it. Can't you just replace the condition with something along `if (Math.Abs(nearestTick - time) <= snapTimePerBeat + 0.1)`? Also, is that 0.1 constant good, or maybe it should be smaller?

Reply:
it might be better to change that value depending on the bpm, i can try some things later